### PR TITLE
Add centrifuge_kreport module

### LINKED ISSUE #833: new module: centrifuge
I think it would be good to have a module for [centrifuge](https://bioconda.github.io/recipes/centrifuge/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

nf-core/mag uses a [local module](https://github.com/nf-core/mag/blob/master/modules/local/centrifuge.nf). Centrifuge is similar to Kraken2, which is already available as [module](https://github.com/nf-core/modules/tree/master/modules/kraken2/kraken2).

### PR DESCRIPTION
**Pull request by sofstam**
## PR checklist

Closes #833  

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/centrifuge/kreport/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    db_name=`find -L ${db} -name "*.1.cf" -not -name "._*"  | sed 's/.1.cf//'`
+    centrifuge-kreport -x \$db_name ${prefix}.results.txt > ${prefix}.kreport.txt
```
Comment (jfy133) [inline modules/centrifuge/kreport/main.nf:30]:
```suggestion
    centrifuge-kreport -x \$db_name {results}.txt > ${results}.txt
```

Also add warning as in point 5 in the guidelines [^1]

[^1]: https://nf-co.re/docs/contributing/modules#naming-conventions

Comment (sofstam) [reply modules/centrifuge/kreport/main.nf:26]: Wouldn't we need the `prefix` here?

The output of centrifuge_centrifuge is: ` -S ${prefix}.results.txt`
So maybe `centrifuge-kreport -x \$db_name ${prefix}.results.txt > ${prefix}.results.txt` ?

File: tests/modules/centrifuge/kreport/test.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+- name: centrifuge kreport test_centrifuge_kreport_single_end
+  command: nextflow run tests/modules/centrifuge/kreport -entry test_centrifuge_kreport_single_end -c tests/config/nextflow.config
+  tags:
+    - centrifuge/kreport
+    - centrifuge
+  files:
+    - path: output/centrifuge/test.kreport.txt
```
[the remark this answers is not in the document]
Comment (jfy133) [reply tests/modules/centrifuge/kreport/test.yml:7]: Is there any keyword or something within the `.txt` file you can use to verify it was generated corretly (i.e. not empty)

for example if th efirst row was like

```
taxon count
```

you could use that string with a [`contains:`](https://pytest-workflow.readthedocs.io/en/stable/#test-options) block

Example: https://github.com/nf-core/modules/blob/e751e5040af57e1b4e06ed4e0f3efe6de25c1683/tests/modules/fastp/test.yml

File: modules/centrifuge/kreport/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/centrifuge/kreport/main.nf:26]: Oops, yes confused myself -  updated the sample.

If output of `centrifuge_centrifuge` is _already_ `<prefix>.results.txt`, te nyou can just specify the variable alone (don't need to reconstruct it with `${prefix}`

Your example above wouldn't work as that would _definitely_ lead to a file name collision (both input and output would be the same), but if we don't specify `kreport`, and just `${prefix}.txt`, this is OK as this doesn't have `results` by default ,as we are creating the file ourselvs.

File: tests/modules/centrifuge/kreport/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sofstam) [reply tests/modules/centrifuge/kreport/test.yml:7]: Yes, sure!

File: modules/centrifuge/kreport/main.nf (hunk 1 above)
Comment (sofstam) [reply modules/centrifuge/kreport/main.nf:26]: Sounds good!

File: tests/modules/centrifuge/kreport/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sofstam) [reply tests/modules/centrifuge/kreport/test.yml:7]: Or not :D 

This is how the file looks like:

[test.kreport.txt](https://github.com/nf-core/modules/files/8479115/test.kreport.txt)

[the remark this answers is not in the document]
Comment (jfy133) [reply tests/modules/centrifuge/kreport/test.yml]: :laughing: 

I guess you could just use `unclassified`?

[the remark this answers is not in the document]
Comment (sofstam) [reply tests/modules/centrifuge/kreport/test.yml]: Yes, I was thinking of using a header if any but `unclassified `works fine.

**Review by jfy133 — APPROVED** [review]

**Pull request merged**